dropped_metrics_test.go

```diff
@@ -0,0 +1,214 @@
+package veneur_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stripe/veneur/v14"
+	"github.com/stripe/veneur/v14/samplers"
+)
+
+type tcData struct {
+	reason  string
+	metrics []samplers.InterMetric
+}
+
+type tcExpected struct {
+	reason string
+	data   map[string]int64
+}
+
+var testCases = []struct {
+	name     string
+	data     []tcData
+	expected []tcExpected
+}{
+	{
+		name: "simple case",
+		data: []tcData{
+			{
+				reason: "simple",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "simple",
+				data: map[string]int64{
+					"foo": 1,
+				},
+			},
+		},
+	},
+	{
+		name: "simple multiple same metric submitted",
+		data: []tcData{
+			{
+				reason: "simple",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "simple",
+				data: map[string]int64{
+					"foo": 3,
+				},
+			},
+		},
+	},
+	{
+		name: "multiple metrics with different reasons",
+		data: []tcData{
+			{
+				reason: "one",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+				},
+			},
+			{
+				reason: "two",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "bar",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "one",
+				data: map[string]int64{
+					"foo": 1,
+				},
+			},
+			{
+				reason: "two",
+				data: map[string]int64{
+					"bar": 1,
+				},
+			},
+		},
+	},
+	{
+		name: "multiple overlapping metrics with different reasons",
+		data: []tcData{
+			{
+				reason: "one",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+					{
+						Name: "bar",
+					},
+				},
+			},
+			{
+				reason: "two",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "bar",
+					},
+					{
+						Name: "bar",
+					},
+					{
+						Name: "bar",
+					},
+					{
+						Name: "foo",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "one",
+				data: map[string]int64{
+					"foo": 3,
+					"bar": 1,
+				},
+			},
+			{
+				reason: "two",
+				data: map[string]int64{
+					"bar": 3,
+					"foo": 1,
+				},
+			},
+		},
+	},
+}
+
+func Test_DroppedMetrics(t *testing.T) {
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
I don't know what the preferred stylistic way to do this is, but I personally prefer creating helper methods and calling them from individual tests. Whatever you think is more maintainable.

Reply:
Unsure whatcha mean, do you mean cutting down on the LoC in the `t.Run()` or moving away from table-driven testing?

Reply:
I mean moving away from table-driven testing. That's my personal preference though, which I don't want to impose into code review. I was thinking creating a separate `Test...(t *testing.T)` for each name you have in the table.